vulkan-cts: 1.3.7.0 -> 1.3.7.2

Flakebi opened the pull request:
## Description of changes

Changelog: https://github.com/KhronosGroup/VK-GL-CTS/releases/tag/vulkan-cts-1.3.7.1
Changelog: https://github.com/KhronosGroup/VK-GL-CTS/releases/tag/vulkan-cts-1.3.7.2
Builds and works.

Also add an impureTest that runs a test on the GPU.


<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

NickCao commented:
We do have lavapipe, the software vulkan implementation, is it possible to run cts on that?

Flakebi commented:
Good idea. Seems to work just fine, I changed to a normal test using lavapipe.

NickCao commented on pkgs/tools/graphics/vulkan-cts/default.nix:
```diff
@@ -34,82 +37,86 @@ let
 
-  passthru.updateScript = ./update.sh;
+    passthru.updateScript = ./update.sh;
+    passthru.tests.run = runCommand "vulkan-cts-lavapipe-run" { nativeBuildInputs = [ pkg ]; } ''
```
You can use `finalAttrs.finalPackage` here.

NickCao commented on pkgs/tools/graphics/vulkan-cts/default.nix:
```diff
@@ -1,6 +1,8 @@
 { lib, stdenv
+, callPackage
```
Unused input.

NickCao commented on pkgs/tools/graphics/vulkan-cts/default.nix:
```diff
@@ -104,6 +107,10 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   passthru.updateScript = ./update.sh;
+  passthru.tests.run = runCommand "vulkan-cts-lavapipe-run" { nativeBuildInputs = [ finalAttrs.finalPackage ]; } ''
```
```suggestion
  passthru.tests.lavapipe = runCommand "vulkan-cts-tests-lavapipe" { nativeBuildInputs = [ finalAttrs.finalPackage ]; } ''
```

NickCao reviewed (commented):
Some nitpicks, overall LGTM.

NickCao commented on pkgs/tools/graphics/vulkan-cts/default.nix:
```diff
@@ -104,6 +106,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   passthru.updateScript = ./update.sh;
+  passthru.tests.lavapipe = runCommand "vulkan-cts-tests-lavapipe" { nativeBuildInputs = [ finalAttrs.finalPackage ]; } ''
+    VK_ICD_FILENAMES=${mesa.drivers}/share/vulkan/icd.d/lvp_icd.x86_64.json deqp-vk -n dEQP-VK.api.smoke.triangle
```
Oh wait, is hardcoding 86_64 intended!

Flakebi replied:
Now using a `*` with some weird bash because `export` does not expand wildcards.

Flakebi committed:
vulkan-cts: 1.3.7.0 -> 1.3.7.2

Changelog: https://github.com/KhronosGroup/VK-GL-CTS/releases/tag/vulkan-cts-1.3.7.1
Changelog: https://github.com/KhronosGroup/VK-GL-CTS/releases/tag/vulkan-cts-1.3.7.2

Also add an test that runs a simple cts test with lavapipe.
